pkg/eventing/v1alpha1/client.go

```diff
@@ -161,8 +161,12 @@ func (b *TriggerBuilder) Subscriber(subscriber *duckv1.Destination) *TriggerBuil
 }
 
 // Broker to set the broker of trigger object
-func (b *TriggerBuilder) Broker(broker string) *TriggerBuilder {
+func (b *TriggerBuilder) Broker(broker string, inject bool) *TriggerBuilder {
```

Comment:
I would add a separate builder method like `"InjectBroker()"` instead of mangling it with the broker. Also, when injection is enabled the broker has to be set to "default" anyway, so there are only limited combinations of this argument possible.